index.js

```diff
@@ -1,9 +1,26 @@
 var sha3 = require("crypto-js/sha3");
 var schema_version = require("./package.json").version;
+var Ajv = require("ajv");
+var contractSchema = require("./spec/contract.spec.json");
 
 // TODO: This whole thing should have a json schema.
 
 var TruffleSchema = {
+  // Return a promise to validate a contract object
+  // - Resolves as validated `contractObj`
+  // - Rejects with list of errors from schema validator
+  validateContractObject: function(contractObj) {
+    return new Promise(function (resolve, reject) {
+      var ajv = new Ajv();
+      var validate = ajv.compile(contractSchema);
+      if (validate(contractObj)) {
+        resolve(contractObj);
+      } else {
+        reject(validate.errors);
+      }
+    });
+  },
```

Comment:
> Still, I like validate() over validateContractObject()

Agreed, I'll switch it.

> Additionally, why is this wrapped inside a promise? It looks as though Ajv's validate function is synchronous, meaning you can have it throw instead of promisified. I recommend only wrapping it in a promise if it's doing asynchronous work or if there's a high likelihood we'll add asynchronous code to the function in the future.

I figured that I didn't want to make any assumptions about the underlying schema validation library. If Ajv turns out to not be the best choice and we have to change it to something that uses Promises, we'd already be covered.

But it might be heavy handed, I'm kind of just playing around to see what looks good / makes sense. Your call!